sw/device/lib/crypto/drivers/kmac.c

```diff
@@ -38,95 +260,208 @@ static kmac_error_t wait_status_bit(uint32_t bit_position, uint32_t bit_value) {
 }
 
 OT_WARN_UNUSED_RESULT
-kmac_error_t kmac_init(kmac_operation_t operation) {
+static kmac_error_t write_left_encode(const size_t len, uint8_t *encoded_len,
+                                      size_t *write_ctr) {
+  if (encoded_len == NULL || write_ctr == NULL || len > kKmacPrefixMaxSize) {
+    return kKmacArgsError;
+  }
+
+  size_t bit_len = len * 8;
+  if (bit_len < 256) {
+    // Need 1 byte to encode `len`
+    *encoded_len++ = 0x01;
+    *encoded_len++ = bit_len;
+    *write_ctr += 2;
+  } else {
+    // Need 2 bytes to encode `len`
+    *encoded_len++ = 0x02;
+    *encoded_len++ = bit_len / 256;
+    *encoded_len++ = bit_len % 256;
+    *write_ctr += 3;
+  }
+  // `bit_len` is upper bounded by kKmacPrefixSize * 8 = 288, so we don't need
+  // to implement larger encoding cases
```

Comment:
Ah, I assumed `kKmacPrefixSize` was a compile-time constant since it starts with `k`. In any case, I see there's now a fall-through error, so this seems resolved.